src/Haskell/Ide/Engine/Plugin/Brittany.hs

```diff
@@ -65,21 +65,28 @@ formatText
   => Maybe FilePath -- ^ Path to configs. If Nothing, default configs will be used.
   -> FormattingOptions -- ^ Options for the formatter such as indentation.
   -> Text -- ^ Text to format
-  -> m (Either [BrittanyError] Text) -- ^ Either formatted Text or a error from Brittany. 
-formatText confFile opts text = 
+  -> m (Either [BrittanyError] Text) -- ^ Either formatted Text or a error from Brittany.
+formatText confFile opts text =
   liftIO $ runBrittany tabSize confFile text
   where tabSize = opts ^. J.tabSize
 
--- | Extend to the line below to replace newline character, as above.
+-- | Extend to the line below and above to replace newline character.
```

Comment:
Does this extend the range to the line above? It looks like it does

```
xxxSooo
ooExxxx
xxx
=>

Soooooo
oooooo
Exxx

```

Reply:
Also this looks quite handy. Would it be worthwhile look at moving these helper functions into `haskell-lsp-types`? cc @alanz

Reply:
It could be handy to do that, but as a follow-up, this PR has had a very long history already, its unfair to ask for that too.

Reply:
I agree, this is out of scope for this PR. I think it would make more sense as part of a refactoring